microsoft-edge/extensions-chromium/developer-guide/minimize-page-load-time-impact.md

```diff
@@ -0,0 +1,260 @@
+
+
+<!-- ------------------------------ -->
+#### Profile your content script using the Edge tracing tool
+
+The Edge tracing tool, available at the `edge://tracing` URL, is a powerful tool that can provide detailed analysis of your extension's performance. In this section, you'll learn how to use the Edge tracing tool to understand the impact of your extension on the page load time. To learn more about this tracing tool, which is based on the **Perfetto** tool, see [Perfetto UI](https://perfetto.dev/docs/visualization/perfetto-ui) in the Perfetto Tracing Docs.
+
+1. To open the Edge tracing tool, open a new tab or window, and go to `edge://tracing`. The tracing UI opens.
+
+1. To start a new trace, in the top left corner of the tool, click the **Record** button. The **Record a new trace** dialog opens.
+
+1. Select the **Manually select settings** option button. The list of categories appears.
+
+1. To capture detailed information about your extension's content script compilation and execution, select all of the following categories:
+
+   * **extensions**
+   * **v8**
+   * **devtools**
+   * **devtools.timeline**
+
+1. Click the **Record** button.  The dialog closes, and the Edge tracing tool starts recording the trace.
+
+1. Open a new tab and load a webpage that your extension affects. The tracing tool collects data about the performance impact of your extension on the webpage.
+
+1. Open the tab where the Edge tracing tool is running, and then click the **Stop** button. The new trace information appears in the tool.
+
+Traces that are recorded by the Edge tracing tool provide a lot of information about the browser, as well as your extension.
+
+To filter the information to show only what's relevant to the webpage that your extension impacted:
+
+1. In the `edge://tracing` page, press **Shift+Esc** to open the **Browser Task Manager** dialog.
+
+1. In the **Browser Task Manager** dialog, search for the tab that corresponds to the webpage that your extension affected, and note the number in the **Process ID** column. Close the dialog.
+
+1. In the toolbar of the Edge tracing tool, click **Processes**, and then select the checkbox that corresponds to the process ID you noted. Clear all other checkboxes.
+
+1. In the upper right corner of the Edge tracing tool, click the search field, type **ScriptInjection::InjectJS**, and then press **Enter** repeatedly until an event that corresponds to your extension is highlighted in the bottom panel.
+
+   The bottom panel displays the start time and total duration of the event:
+
+   ![The Edge tracing tool showing a script injection event](./minimize-page-load-time-impact-images/tracing-event-selected.png)
+
+To continue analyzing the performance impact of your extension's content script on the webpage, look for the following key events inside the **ScriptInjection::InjectJS** event:
```

Comment:
does this set of lines apply to both tools? Microsoft Edge DevTools & Edge tracing tool

affects h4 vs h6

current headings structure implies that this implicit section applies only to Edge tracing tool, not to DevTools, but it seems like it could apply to DevTools as well thus this would be h4 not h6

rendered: just above the heading:
https://review.learn.microsoft.com/en-us/microsoft-edge/extensions-chromium/developer-guide/minimize-page-load-time-impact?branch=pr-en-us-3272#only-add-content-script-code-thats-needed-for-the-functionality-of-your-extension

@andrejo-msft @captainbrosset

Reply:
tracing tool only therefore h6 not h4

Reply:
My understanding is that it only applies to the Edge tracing tool.